lib/classes/whereable.rb

```diff
@@ -0,0 +1,16 @@
+module Whereable
+  attr_accessor :relation
+
+  def use_model(mod)
+    @model = mod
+    @relation = mod.all
```

Comment:
I think using `.all` is risky if table has a large number of rows. `.all` instructs Active Record to fetch the entire table in a single pass and it stores all records in memory which is not required here as per my understanding.

Method used here : https://github.com/crossroads/api.goodcity/pull/840/files#diff-ed6a25795ba8774bf0e253b8bfab146aR34

Which call https://github.com/crossroads/api.goodcity/pull/840/files#diff-ed6a25795ba8774bf0e253b8bfab146aR79

It looks like we are trying something similar to 
`PackagesInventory.all.where("#{model.class.name.underscore}_id = (?)", Utils.to_id(model))`

which can be done as `PackagesInventory.where("#{model.class.name.underscore}_id = (?)", Utils.to_id(model))`

correct me If I am wrong. :)

Reply:
All is just a base relation which gets extended. If it does indeed get computed instantly then I'll just replace it with `where(nil)`. Will double check